src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -332,124 +306,162 @@ function scprintref(cb)
-        )
-        println(
-            io,
-            "#  [         :                :          :        :      :          :           ],",
-        )
-        println(io, "# ]")
-        #
-        # Write tables
-        #  Reference value and precision match tables are separate since it is more
-        #  common to update reference values occasionally while precision values are
-        #  typically changed rarely and the precision values are hand edited from experience.
-        #
-        # Write table of reference values
-        println(io, "varr = [")
-        for lv in sc.cur_stats_flat
-            s1 = lv[1]
-            l1 = length(s1)
-            s1 = sp[1:(a1l - l1)] * "\"" * s1 * "\""
-            s2 = lv[2]
-            if typeof(s2) == String
-                l2 = length(s2)
-                s2 = sp[1:(a2l - l2)] * "\"" * s2 * "\""
-                s22 = ""
-            end
-            if typeof(s2) == Symbol
-                s22 = s2
-                l2 = length(String(s2))
-                s2 = sp[1:(a2l - l2 + 1)] * ":"
+    mpirank = MPI.Comm_rank(MPI.COMM_WORLD)
+    mpirank == 0 || return
+    # Get print format lengths for cols 1 and 2 so they are aligned
+    # for readability.
+    phi = sc.cur_stats_flat
+    a1l = maximum(length.(map(x -> x[1], phi)))
+    a2l = maximum(length.(String.(map(x -> x[2], phi))))
+    fmt1 = @sprintf("%%%d.%ds", a1l, a1l) # Column 1
+    fmt2 = @sprintf("%%%d.%ds", a2l, a2l) # Column 2
+    fmt3 = @sprintf("%%28.20e")         # All numbers at full precision
+    # Create an string of spaces to be used for formatting
+    sp = repeat(" ", 75)
+
+    # Write header
+    println(io, "# BEGIN SCPRINT")
+    println(io, "# varr - reference values (from reference run)    ")
+    println(io, "# parr - digits match precision (hand edit as needed) ")
+    println(io, "#")
+    println(io, "# [")
+    println(
+        io,
+        "#  [ MPIStateArray Name, Field Name, Maximum, Minimum, Mean, Standard Deviation ],",
+    )
+    println(
+        io,
+        "#  [         :                :          :        :      :          :           ],",
+    )
+    println(io, "# ]")
+    #
+    # Write tables
+    #  Reference value and precision match tables are separate since it is more
+    #  common to update reference values occasionally while precision values are
+    #  typically changed rarely and the precision values are hand edited from experience.
+    #
+    # Write table of reference values
+    println(io, "varr = [")
+    for (s1, s2, s3′, s4′, s5′, s6′) in sc.cur_stats_flat
+        s1 = sp[1:(a1l - length(s1))] * "\"" * s1 * "\""
+        s2 = sp[1:(a2l - length(s2))] * "\"" * s2 * "\""
+        s3 = sprintf1(fmt3, s3′)
+        s4 = sprintf1(fmt3, s4′)
+        s5 = sprintf1(fmt3, s5′)
+        s6 = sprintf1(fmt3, s6′)
+        srow = (s3, s4, s5, s6)
+        println(io, " [ ", s1, ", ", s2, join(srow, ","), " ],")
```

Comment:
I think this line needs another `", ",` after `s2`

Example output after rebasing onto to master: 
```
varr = [
 [ "state",     "ρ"  9.69901051313856066294e-01,  1.00695111308847851106e+00,  1.00000002953972089159e+00,  6.94722590819101589593e-03 ],
 [ "state", "ρu[1]" -4.21771768146742886962e-01,  6.34790159642324547384e-01,  1.59384804664454482470e-01,  2.22844788492525480716e-01 ],
 [ "state", "ρu[2]" -4.37678540756429201863e-01,  4.53959896024816400573e-01, -1.22554654623556756642e-04,  1.73930782418124457722e-01 ],
 [ "state",    "ρθ" -1.13636401173616619076e+00,  1.51145239791767727056e+00,  1.96870293018723595616e-03,  3.82573373360323043535e-01 ],
]
```

should I open up a PR for this one line fix? @charleskawczynski

Reply:
Ah, yes, sorry about that. That would be great, thank you!

Reply:
I'm surprised I didn't see the diff from this. Are these ref values git-ignored?